File: 2023m./uzd1./sprendimai/uzd1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>

using namespace std;

class Cls {
public:
    int studentCount = 1;
    int category;
    double kmWalked;

    Cls(int category, double kmWalked) : category(category), kmWalked(kmWalked) {}
};

int main() {
    ifstream data("U1.txt");
    ofstream output("U1rez.txt");

    vector<Cls> classes;

    int numStudents;
    data >> numStudents;

    for (int i = 0; i < numStudents; ++i) {
        int category, stepDistance;
        data >> category;
        data >> stepDistance;
        int cmWalked = 0;
        bool hasUnregisteredDay = false;

        for (int j = 0; j < 7; ++j) {
            int daySteps;
            data >> daySteps;
            if (daySteps == 0) {
                cmWalked = 0;
                hasUnregisteredDay = true;
            }
            cmWalked += daySteps * stepDistance;
        }
        double kmWalked = 0;

        if(!hasUnregisteredDay) kmWalked = round(cmWalked * 0.001) * 0.01;

        if (kmWalked > 0) {
            bool foundClass = false;
            for (Cls& cls : classes) {
                if (cls.category == category) {
                    cls.kmWalked += kmWalked;
                    cls.studentCount++;
                    foundClass = true;
                    break;
                }
            }
            if (!foundClass) {
                classes.push_back(Cls(category, kmWalked));
            }
        }
    }

    for (const Cls& cls : classes) {
        output << cls.category << " " << cls.studentCount << " " << cls.kmWalked << endl;
    }

    data.close();
    output.close();

    return 0;
}

File: 2023m./uzd1./sprendimai/uzd1.py
class Cls:
    studentCount = 1
    def __init__(self, category, kmWalked):
        self.category = category
        self.kmWalked = kmWalked

def main():
    data = open("U1.txt", "r")
    output = open("U1rez.txt", "w")


    classes = []

    for i in range(int(next(data))):
        studentData = next(data).split(" ")
        studentData = [int(o) for o in studentData]
        cmWalked = 0
        for j in range(2, 9):
            if(studentData[j] == 0):
                cmWalked = 0
                break
            cmWalked += studentData[j] * studentData[1]
        if(cmWalked == 0):
            continue
        kmWalked = round(cmWalked * 0.00001, 2)
        foundClass = False
        for j in range(len(classes)):
            if(classes[j].category == studentData[0]):
                classes[j].studentCount += 1
                classes[j].kmWalked += kmWalked
                foundClass = True
                break
        if(not foundClass):
            classes.append(Cls(studentData[0], kmWalked))

    for cls in classes:
        output.write(f"{cls.category} {cls.studentCount} {cls.kmWalked}\n")

    data.close()
    output.close()

if __name__ == "__main__":
    main()